api/apps/v1alpha1/nemo_common_types.go

```diff
@@ -47,32 +47,99 @@ type DatabaseConfig struct {
 	// DatabaseName is the database name for a NEMO Service.
 	// Required, must not be empty.
 	//
-	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:MinLength=1
-	DatabaseName string `json:"databaseName,omitempty"`
+	DatabaseName string `json:"databaseName"`
 	// DatabaseCredentials stores the configuration to retrieve the database credentials.
 	// Required, must not be nil.
 	//
-	// +kubebuilder:validation:Required
-	Credentials *DatabaseCredentials `json:"credentials,omitempty"`
+	Credentials *DatabaseCredentials `json:"credentials"`
 }
 
+// DatabaseCredentials are the external database credentials
 type DatabaseCredentials struct {
 	// User is the non-root username for a NEMO Service in the database.
 	// Required, must not be empty.
 	//
-	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:MinLength=1
-	User string `json:"user,omitempty"`
+	User string `json:"user"`
 	// SecretName is the name of the secret which has the database credentials for a NEMO service user.
 	// Required, must not be empty.
 	//
-	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:MinLength=1
-	SecretName string `json:"secretName,omitempty"`
+	SecretName string `json:"secretName"`
 	// PasswordKey is the name of the key in the `CredentialsSecret` secret for the database credentials.
 	// Defaults to "password".
 	//
 	// +kubebuilder:default:="password"
 	PasswordKey string `json:"passwordKey,omitempty"`
 }
+
+// WandBSecret represents the secret and key details for the Weights and Biases service.
+type WandBSecret struct {
+	// Name is the name of the Kubernetes Secret containing the WandB API key.
+	// Required, must not be empty.
+	//
+	// +kubebuilder:validation:MinLength=1
+	Name string `json:"name"`
+
+	// APIKeyKey is the key in the Secret that holds the WandB API key.
+	// Defaults to "apiKey".
+	// +kubebuilder:default="apiKey"
+	// +kubebuilder:validation:MinLength=1
+	APIKeyKey string `json:"apiKeyKey"`
```

Comment:
Nit: APIKey should be good

Reply:
@slu2011 that might cause confusion and users might provide the actual api key as input than the key representing it in the secret